.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,59 @@
+name: Continuous integration reusable workflow
+on:
+  workflow_call:
+    inputs:
+      continue-on-error:
+        required: true
+        type: string
+      build_args:
+        type: string
+      dockerfile:
+        type: string
+      tag:
+        type: string
+
+jobs:
+  build-and-push-candidate:
+    continue-on-error: ${{ inputs.continue-on-error == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout out code
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v2
+        if: github.repository_owner == 'kokkos' && ( github.event_name == 'push' ||  github.event_name == 'schedule' )
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Convert build_args
+        id: build_args
+        run: |
+          echo "args<<EOF" >> $GITHUB_OUTPUT
+          echo ${{ inputs.build_args }} | sed 's/,/\n/' >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+      - name: Build and Push Docker images for Github Container Registry
```

Comment:
TODO: check `outputs` option to see if the built image can be pushed locally, tested in `test-kokkos` and only then pushed to registry in `push-stable`.

Reply:
For reference: https://docs.docker.com/build/ci/github-actions/test-before-push/